### PR TITLE
Features/job runs index

### PR DESCRIPTION
**Pull request by rupurt**
[Finishes #157751425]

### CODE REVIEW

**Comment by rupurt** [comment]
@se3000 I added a story to the icebox to try an silence those 404 route requests for dynamic routes https://www.pivotaltracker.com/story/show/158299642

Can you create a quicktime video of the other error that you're running into?

File: gui/__tests__/containers/JobSpec.test.js [hunk 1]
```diff
@@ -40,5 +41,30 @@ describe('containers/JobSpec', () => {
     expect(wrapper.text()).toContain('Created2018-05-10T00:41:54.531043837Z')
     expect(wrapper.text()).toContain('Run Count1')
     expect(wrapper.text()).toContain('{"value":"8400.00"}')
+    expect(wrapper.text()).not.toContain('Show More')
+  })
+
+  it('displays a show more link if there are more than 5 runs', async () => {
+    const runs = [
```
Comment (j16r) [inline gui/__tests__/containers/JobSpec.test.js]: Could we use the injection trick here to reduce the verbosity of this test?

File: gui/support/test-helpers/clickNextPage.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export default wrapper => {
+  wrapper = wrapper.update() // Fixes enzyme finder bug
```
Comment (j16r) [inline gui/support/test-helpers/clickNextPage.js]: Got a link to this bug / issue on github / on stack overflow? What is the bug?

File: gui/src/containers/JobSpec.js [hunk 3]
```diff
@@ -84,13 +92,20 @@ const renderJobSpec = ({classes, jobSpec, latestJobRuns, jobRunsCount}) => (
...
   <React.Fragment>
     <Typography variant='title' className={classes.lastRun}>
       Last Run
     </Typography>
 
-    <JobRunsList runs={latestJobRuns} />
+    <Card>
+      <JobRunsList runs={latestJobRuns} />
+    </Card>
+    {jobRunsCount > LATEST_JOB_RUNS_COUNT &&
```
Comment (j16r) [inline gui/src/containers/JobSpec.js]: Is there not an `{if ...}markup{end}` syntax we could use here?

File: gui/__tests__/containers/JobSpec.test.js (hunk 1 above)
Comment (rupurt) [reply gui/__tests__/containers/JobSpec.test.js]: @j16r this one is a higher level integration test.

File: gui/support/test-helpers/clickNextPage.js (hunk 2 above)
Comment (rupurt) [reply gui/support/test-helpers/clickNextPage.js]: https://github.com/airbnb/enzyme/issues/1233#issuecomment-385343903

File: gui/src/containers/JobSpec.js (hunk 3 above)
Comment (rupurt) [reply gui/src/containers/JobSpec.js]: This is how you do inline `if` in JSX https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

The other option is to pull it out into a variable but I prefer this style.

File: gui/support/test-helpers/clickNextPage.js (hunk 2 above)
Comment (j16r) [reply gui/support/test-helpers/clickNextPage.js]: This should be in the code ;)

**Pull request merged**